### DISCUSSION TITLE
GT 4286 1619 games

### DISCUSSION THREAD

**Discussion by ExerionExo** [nk64/GT4286#5] [General]
Hi, Hope this is ok. To help others out that may have a corrupted SD card out of the box like one of mine, I'm in the process of uploading a zip archive of the 1619 game card to the internet archive. The file just needs to be unzipped onto a SD card. It's going to be slow as I'm fixed wireless. Size is approx. 8.5GB

**Comment by jacksonho81** [comment]
Sounds great please let us know once available

**Reply by ExerionExo** [reply]
Will do

**Reply by ExerionExo** [reply]
https://archive.org/details/gt-4286-1619-games

**Reply by jacksonho81** [reply]
Hey, it took me a week to get around to this and noticed the file is now gone. Did you upload it anywhere else? I assume archive.org removed it for copywrite?

**Reply by horsiezoo** [reply]
Ah so it has probably because of all the pirated stuff on that, tell you what when I’m free I’ll upload an image stripped of all roms as a plus it’ll be a lot smaller. Or: was it the roms you were after? In that case Google = friend

**Reply by jacksonho81** [reply]
I'm just getting around to reviewing the discussion from the last 2 weeks. I got the 150 game version, can I just get some MAME roms and drop them into the /roms/MAME folder?

**Reply by horsiezoo** [reply]
Yes read the main (code) pages first then see if you can follow the discussion threads from the beginning where everyone was trying to figure out “where everything goes” also feel free feel to add anything/ correct anything as you go. Maybe copy paste stuff from threads into notepad as it’s kind of all over the place

**Reply by jacksonho81** [reply]
Okay thanks for that, trying to read through. I was hoping I could just drop the MAME / NES files into the ROMS drives and they would all appear, doesn't appear to be that simple, trying to see how to reload the ROM list into the GUI.

**Reply by jacksonho81** [reply]
Got it sorted, downloaded the tool

**Reply by horsiezoo** [reply]
Good stuff. Heh so you are discovering that basically everything printed on the box is untrue right?
No idea how Jaycar can get away with that on top of removing all the open source licenses from the software they used :-)

**Comment by cgb** [comment]
Do you know if the 1619 game card is compatible with the GT-4286 ?  i.e. is it the same hardware for both models?

**Reply by Hello-dot-Jpg** [reply]
Sorry I was wrong it is

**Comment by Hello-dot-Jpg** [comment]
Hi Guys, I just realized something, if you have the OG SD card and have NOT copied anything on to it you could probably run a program like PhotoRec to recover the removed ROMs, my guess is that they just deleted the roms and regenerated the database on the master SD card before sticking it in the cloning machine

If you haven't read the info about the images directories, there seems to be a lot more images than there are available ROMs which makes me think they removed a whole bunch for copywrite reasons.

**Reply by Hello-dot-Jpg** [reply]
I am running a scan on mine it's found a bunch of stuff but I had copied files onto it before so I am not sure if it will be able to recover the files

**Reply by cgb** [reply]
If anyone is intending to do forensic analysis, consider capturing the SD card to a image file using ddrescue or similar... Then perform the analysis on the image file.

**Reply by nk64** [reply]
There are a lot of vestigial files on my SD card that hint at the previous existence of other roms:

- 3054 ```.cfg``` files in ```/emus/fceux/cfg```
- 366 ```.sav``` files in ```/emus/fceux/sav```
- 175 ```.md.srm``` files in ```/emus/picodrive/srm```
- 4028 ```.cfg``` files in ```/emus/snes9x4d```
- 1625 ```.srm``` files in ```/emus/snes9x4d/snapshots```

Over the next couple of days I will be working on a command line utility to allow you to add the roms you want to the system as if they were built in roms.

I understand that for many the idea of copying an existing SD card flush with games is satisfactory, but for those who want some more control over their roms I hope that my tool will help with that.

**Reply by Hello-dot-Jpg** [reply]
I'm considering purchasing another one, if I do, I'll image the SD card before changing anything then run a full scan and recovery of the free space, hopefully that will restore the removed roms, if it works, I'll upload it to the internet archive

**Reply by Hello-dot-Jpg** [reply]
From my recovery it does seem like the Roms are on there just that they have been deleted at some point

![f57107776](https://github.com/user-attachments/assets/7f514173-a8cc-42be-8f24-ee471605e9c6)

Is that from the UI on the 1619 version?

**Comment by ExerionExo** [comment]
Hi,

Yes, my console is the GT 4286. Apparently Jaycar had to change the roms due to copyright issues. Some of us got the older systems with the larger rom set. A lot of good games though some are non-English and near impossible to play. My upload is up to 1.9GB out of 8.3GB atm, very slow upload speed. Cheers Geoff

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________
From: cgb ***@***.***>
Sent: Friday, December 27, 2024 6:42:50 pm
To: nk64/GT4286 ***@***.***>
Cc: Geoff Burt ***@***.***>; Author ***@***.***>
Subject: [External][MARKETING/BULK]Re: [nk64/GT4286] GT 4286 1619 games (Discussion #5)


CAUTION: This email originated from outside of TAFE NSW. Do not click links or open attachments unless you recognise the sender and know the content is safe.


Do you know if the 1619 game card is compatible with the GT-4286 ? i.e. is it the same hardware for both models?

—
Reply to this email directly, view it on GitHub<https://github.com/nk64/GT4286/discussions/5#discussioncomment-11675631>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BOABLKD2YM4Y3CX7YFQYTOL2HUAHHAVCNFSM6AAAAABUIC7CNWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRXGU3DGMI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

***** The contents of this email and its attachments are confidential and intended solely for the use of the individual or entity to whom they are addressed. *****

**Reply by ExerionExo** [reply]
https://archive.org/details/gt-4286-1619-games

**Reply by horsiezoo** [reply]
Thank you for the upload!

**Reply by Hello-dot-Jpg** [reply]
Great work

**Reply by nk64** [reply]
@ExerionExo Thanks for the upload, your SD Card is slightly different to the other one is available by DM on ozbargain. Yours actually has 1619 rom files but the other is missing 7 rom files.

I have now found 4 generations of SD Card:

- 2023-05-25: 1619 Games
    - /ui/gamelist.txt contains 1619 lines
    - /game.db contains 1619 builtin games
    - /roms contains a total of 1619 files

- 2023-06-01: 1612 Games (7 roms files removed, no change to menu and emulators)
    - /ui/gamelist.txt contains 1619 lines
    - /game.db contains 1612 builtin games
    - /roms contains a total of 1612 files

- 2024-05-26: 157 Games (Updated Menu and Emulators)
    - /ui/gamelist.txt contains 157 lines
    - /game.db contains 157 builtin games
    - /roms contains a total of 157 files

- 2024-07-22: 157 Games (1 PS1 Game has been renamed to remove a full stop in the name)
    - /ui/gamelist.txt contains 157 lines
    - /game.db contains 157 builtin games
    - /roms contains a total of 157 files

**Reply by Hello-dot-Jpg** [reply]
So does that mean the old model is indeed compatible?

**Reply by psmedley** [reply]
appears mine is the 1619 roms version :)

**Comment by horsiezoo** [comment]
On googling around, miyoo mini handheld is based on same chip,
https://github.com/linux-chenxing/linux-chenxing.org/discussions/41

Long shot but wonder if can extract the more up to date emulators from its firmware, there’s a lot on archive.org and replace. Unfortunately I’m not feeling well today so I’ll see what’s what in next few days. 
Thanks for creating this git page ! I randomly popped into jaycar yesterday while waiting for a dr appointment and saw these monstrosity’s on sale for $60 NZ so here we are. 

SOC wise I suspect it’s 1 gigabit /8 =128MB of ram which seems plausible, and appears to feature in a few cheap aliexpress “smart video projectors”

**Reply by Hello-dot-Jpg** [reply]
Have a look at some of the other discussions that @nk64 and I have been having

**Reply by nk64** [reply]
Sorry to hear that you are not feeling well, but please report back if you have any luck finding binaries to transplant from another system to this one.

**Reply by horsiezoo** [reply]
Hah, so you have! anyways thats a tomorrow thing I think.

I got Tekken 3 running on pcx4all with reasonable performance by editing pcsx.cfg, I think enabling new_dynarec_hacks = 1 does a fair amount of heavy lifting here, to enable cd audio - Cdda = 1

Bios = SCPH1001.BIN
Gpu3 = builtin_gpu
Spu = builtin_spu
Xa = 0
Mdec = 0
Cdda = 1
Debug = 0
PsxOut = 0
SpuIrq = 0
RCntFix = 0
VSyncWA = 0
Cpu = 0
region = 1
g_scaler3 = 2
g_gamma = 64
g_layer_x = 0
g_layer_y = 0
g_layer_w = 280
g_layer_h = 1e0
soft_filter = 0
scanlines = 0
scanline_level = 14
plat_target.vout_method = 0
plat_target.hwfilter = 0
plat_target.vout_fullscreen = 0
state_slot = 0
cpu_clock = ffffffff
g_opts = 1
in_type_sel1 = 0
in_type_sel2 = 0
analog_deadzone = 32
memcard1_sel = ffffffff
memcard2_sel = ffffffff
g_autostateld_opt = 0
adev0_is_nublike = 0
adev1_is_nublike = 0
frameskip3 = 1
gpu_peops.iUseDither = 0
gpu_peops.dwActFixes = 80
gpu_unai.lineskip = 0
gpu_unai.abe_hack = 0
gpu_unai.no_light = 0
gpu_unai.no_blend = 0
gpu_neon.allow_interlace = 2
gpu_neon.enhancement_enable = 1
gpu_neon.enhancement_no_main = 0
gpu_peopsgl.bDrawDither = 0
gpu_peopsgl.iFilterType = 0
gpu_peopsgl.iFrameTexType = 0
gpu_peopsgl.iUseMask = 0
gpu_peopsgl.bOpaquePass = 0
gpu_peopsgl.bAdvancedBlend = 0
gpu_peopsgl.bUseFastMdec = 1
gpu_peopsgl.iVRamSize = 40
gpu_peopsgl.iTexGarbageCollection = 1
gpu_peopsgl.dwActFixes = 0
spu_config.iUseReverb = 1
spu_config.iXAPitch = 0
spu_config.iUseInterpolation = 1
spu_config.iTempo = 0
spu_config.iUseThread = 1
config_save_counter = 1
in_evdev_allow_abs_only = 0
volume_boost = 0
psx_clock = 39
new_dynarec_hacks = 1
in_enable_vibration = 0

**Reply by horsiezoo** [reply]
I replaced the menu music ui/bg_music.wav , simply convert a short audio file to .wav and rename to bg_music.wav and copy over existing file

**Reply by nk64** [reply]
I noticed that your Bios (```SCPH1001.BIN```) is different to the stock SD Card (```SCPH-70004_BIOS_V12_PAL_200 - Kopia.ROM1```). I'm not that familiar with PSX stuff, was there anything important to that change?

**Reply by horsiezoo** [reply]
SCPH-70004_BIOS_V12_PAL_200 - Kopia.ROM1 is a PS2 bios, SCPH1001.BIN is the original PSX bios and is likely the most PSX game compatible. 
PS2 has PSX backwards compatibility but seems unnecessary in this case, /pcsx4all/bios has a lot of PS2 related bios files, I maintain this is definitely a wierd machine.

**Reply by horsiezoo** [reply]
Hmm actually.... ignore all this, I deleted all files in ../pcsx4all/bios  and ... still works, so config and and bios files stored on system flash. That's going to make it harder to make significant changes , durn. :-) (I suspect I may have confused getting better performance from .cfg file tweaks with using a faster sd card. lol, I'll see myself out.)
 Also tried replacing emu files with various ARM versions eg: miyoo mini https://drive.google.com/drive/folders/1diProUuOoVs7nHj0D4L8Kfdicr3Hu6Tm ,among others,  just "crashes" to UI. 
Actually on that, the shell scripts - logs? Would that be of any help here? My understanding is you rename a .sh to the name of an emulator and open that on the console ? 

Messed with "joystick.cfg" in root of sd, (google translate)
joystick老 = joystick old = works same as default
joystick-生产公用 = joystick-production common =  works same as default
joystick无线 = joystick wireless = stick only no buttons work 
joystick新 = joystick new = stick only no buttons work 

So ... my thinking is if can figure out button assignments to enable the "menu" for emus such as gambatte / pcsx4all to change aspect ratio , turn off FPS display etc. I think default on a normal system running gambatte is hold down R,L and select, but on this they are assigned differently (pls tell me I'm wrong here and everything is hard coded in flash)

**Reply by horsiezoo** [reply]
Further experiments : .../emus/fbneo  fbneo-双打 = "doubles" google translate - rename that to fbneo and replace existing - all buttons seem to work without patching, I had a look at its hex and its different to supplied original fbneo and fbneo - 四打 "four dozen" 
All roms that don't work on original don't work on this either, so it's not a newer version ? Unsure.

**Reply by nk64** [reply]
Google translate gave me this:
双打: Doubles
四打: Quadruple

I had originally thought that those characters might indicate 32 bit and 64 bit versions. But they are all 32 bit binaries according to Ghidra.  
I wonder if maybe they mean Two Player/Four Player?
Here are all the references to those characters on my SD Card:
- /emus/fbneo/fbneo - 四打
- /emus/fbneo/fbneo-双打
- /emus/mame/fbneo - 四打
- /emus/mame/fbneo-双打
- /lib/libSDL-1.2.so双打.0
- /lib/libSDL-1.2.so四打.0

You might try renaming the corresponding libSDL-1.2.*.so file to match the emulator version for science.

It is interesting that the doubles one has all buttons mapped through. I might compare the mapping table later.

**Reply by nk64** [reply]
> I deleted all files in ../pcsx4all/bios and ... still works, so config and and bios files stored on system flash.

There is nothing of relevance to the emulation on the flash (I will try and write up how I explored the flash and dumped the contents sometime today)

**Reply by horsiezoo** [reply]
Huh interesting, yes please that would be great! I guess you used a .sh with cp to a folder? Now I’m curious how the psx emulator still works without a bios file. Yes I renamed the .so files, predictably no noticeable difference.

**Reply by nk64** [reply]
> I guess you used a .sh with cp to a folder

Check out the new [section](https://github.com/nk64/GT4286/blob/main/Exploring%20the%20OS.md).
Let's continue the discussion about the contents of the OS [here](https://github.com/nk64/GT4286/discussions/12)

**Reply by nk64** [reply]
@horsiezoo
> Actually on that, the shell scripts - logs? Would that be of any help here? My understanding is you rename a .sh to the name of an emulator and open that on the console ?

Please see the [Capturing emulator output](https://github.com/nk64/GT4286/discussions/13) and let me know how you go.

**Reply by Hello-dot-Jpg** [reply]
> Google translate gave me this: 双打: Doubles 四打: Quadruple
> 
> I had originally thought that those characters might indicate 32 bit and 64 bit versions. But they are all 32 bit binaries according to Ghidra. I wonder if maybe they mean Two Player/Four Player? Here are all the references to those characters on my SD Card:
> 
> * /emus/fbneo/fbneo - 四打
> * /emus/fbneo/fbneo-双打
> * /emus/mame/fbneo - 四打
> * /emus/mame/fbneo-双打
> * /lib/libSDL-1.2.so双打.0
> * /lib/libSDL-1.2.so四打.0
> 
> You might try renaming the corresponding libSDL-1.2.*.so file to match the emulator version for science.
> 
> It is interesting that the doubles one has all buttons mapped through. I might compare the mapping table later.

This would make sense, some of the Arcade game I remember being four player were only running in two player mode (I think there were some cabinets that were two player while others were four)

**Comment by horsiezoo** [comment]
I got Sega MegaCD games working, put bin/cues in ../MD I used scatter shot approach to bios location, but I *think* bios_CD_E.bin bios_CD_J.bin bios_CD_U.bin should be in emus/picodrive I put them in ./bios and the roms/MD folder as well, this needs confirmation via trial /error.
